mitodl/mitxpro
# feat: mimic Learn more feat for Internal courseware as well

#### Pre-Flight checklist


#### What are the relevant tickets?
#2613 

#### What's this PR do?
- Takes the `external_marketing_url` out of the `CourseRun` and into the Course/Program CMS Page. This means moving onward, a course or program page can only have one external marketing URL and not multiple URLs.
- Enables the product page to take the users to 3rd party URLs even for internal courses and programs
- Makes associated API changes with the nesting and attribute `external_marekting_url` location change. This would mean that the `external_marketing_url` would now be a level up in the `api/courses` and `api/programs` response.

#### How should this be manually tested?
- Setup on Master, Setup a couple of external courses ad programs with `external_marketing_url` and make sure you can see the external URL on Learn More button
- Create some internal courses and programs as well or run `./manage.py seed_data` this should create some for you.
- Now, check out this branch, Run migrations and check the migration ran successfully
- Check the external marketing url from the course and program runs has been migrated to the associated course or program page
- Check that the `Learn More` button takes the user to the right URL
- Now the best part is, Add `External marketing URL` value in the internal course and program pages in CMS
- Open the details page of these courses and programs, You should see `Learn More` button for internal courses and programs as well instead of enrolling now. However, the internal courses and programs should be enrollable manually or through an enrollment code.
- Also, check `/api/courses` and `/api/programs` and notice that the `external_marketing_url` field is present under the course or program model structure.

**Also** Check that the CI checks for the JS pass because they were failing because of `caniuse-lite`

**Note** After this PR:

The `external_marketing_url` will take precedence moving onward, This means that if a course or program page has value for the external url field in CMS, we will always show `Learn More` button to the users on the details page.

#### Screenshots (if appropriate)
Internal Program (Details page)
<img width="1610" alt="image" src="https://user-images.githubusercontent.com/34372316/232769543-eccb3745-dc57-4a0d-ab53-1acd7baff2d6.png">
Internal Program (CMS page)
<img width="1594" alt="image" src="https://user-images.githubusercontent.com/34372316/232769798-791208fc-3d6a-4f91-a8a9-c769cd5d14ef.png">

**Internal Course**
<img width="1784" alt="image" src="https://user-images.githubusercontent.com/34372316/233360408-52c7310d-98db-4d3c-a64b-bd74d3265c65.png">
**Internal Course CMS Page**
<img width="1595" alt="image" src="https://user-images.githubusercontent.com/34372316/232770042-8163e459-d645-4ac2-a18c-6697c6513622.png">


**API Response  (Courses)**
<img width="1158" alt="image" src="https://user-images.githubusercontent.com/34372316/233359853-58c100b4-6462-43a9-8027-a6944bf4bff0.png">

**API Response Programs**

<img width="748" alt="image" src="https://user-images.githubusercontent.com/34372316/233360020-cd6be07b-7ca4-4e0a-84be-68e1a1102761.png">


#### What GIF best describes this PR or how it makes you feel?
(Optional)
